arangod/Graph/Enumerators/WeightedTwoSidedEnumerator.h

```diff
@@ -198,9 +277,21 @@ class WeightedTwoSidedEnumerator {
 
     auto getDiameter() const noexcept -> double { return _diameter; }
 
-   private:
-    auto clearProvider() -> void;
+    auto haveSeenOtherSide() const noexcept -> bool {
+      return _haveSeenOtherSide;
+    }
+
+    auto setForbiddenVertices(std::unique_ptr<VertexSet> forbidden)
+        -> void requires HasForbidden<PathValidatorType> {
```

Comment:
Could be more readable to move this requirement up to the template definition in line 133, such that one can directly see all the requirements on all used template parameters.

Reply:
No, this isn't a global requirement. This magic only sees to it that this particular method is present if and only if the specified `PathValidatorType` has a corresponding method. This is because we need to pass the forbidden vertices (and edges) through the Enumerator down to the actual `PathValidators` in the `_left` and `_right` balls.

Reply:
And some (indeed most) instanciations of this template use a `PathValidatorType` which does not have this method.